### DISCUSSION TITLE
Various Improvements

### DISCUSSION THREAD

**Discussion by WalmartWhiteCat** [ppy/osu#25379] [Ideas] [answered]
I've been brainstorming changes that could be made to the game and I think I have a few. They aren't perfect, obviously, I'm not a game dev or anything like that, but I feel they'll be beneficial overall. This might be a little long, with two small additions and two large ones. The small ones are mostly just personal things I think would be nice, but the long ones are where the real meat of this post lies, the ones I think include changes that would fully benefit the game and not just me and a handful of other people.

1 - Sorting by Favourite. Self explanatory, I think this would be nice for when people just want to play their favourite maps.

2 - Skin tab. Make a tab that, when clicked, opens the Skinship Compendium or even a custom made panel of all skins submitted on the skinning forums, like a neat little list or something. This way you can get skins just as easily as you can get maps. You can also make this a Supporter benefit, but that's up to Peppy and whoever else calls the shots.

3 - Skin Editor. As of now, the skin editor is decent, but I can see a few ways to make it better. For one, adding a grid-snap option, like there is in mapping, would be great. This way you can make easier, more precise changes to the placement of elements. Next, make it so when you add an element to the game, it immediately appears where the default element would be. That would also make it easier for skinners to line things up. Finally, I think it would be a great addition to add an option to hide the editor UI, but still make it so elements can be moved when the UI is hidden. I often find myself wanting to do something specific with certain elements, but to do said thing becomes incredibly janky because I have to do it behind the UI. An example of this would be making it so only the, we’ll call it, “Remaining time,” part of the Argon song progress bar is visible. Like, a little timer in the corner. I really like that kind of stuff. But I have to mess with it a ton behind the UI because I can’t hide said UI. And to build off of this, make it so certain parts of the game layout aren’t squished together when the editor is up, this makes it annoying to determine where an element will end up after you close the editor, since some elements shift slightly, making precise changes difficult to perform.

4 - Multiplayer. It’s good for the most part, except for when people choose a difficulty that is incredibly hard just for fun. Or, when there’s a lobby full of people playing your favourite songs, but only the 2 star versions, which are very boring after a certain skill level is reached. I propose implementing a program that will check each player’s individual recommended difficulty, then find a map in the current set that falls the closest to that difficulty. If the difficulty chosen by the program has too much of a time discrepancy compared to the other maps people are playing, we’ll say 30 - 45 seconds, it chooses the next diff below or above depending on the player’s rec-diff (recommended difficulty), and it keeps doing that until there is no sizeable discrepancy or until it hits the same map in the set that the queue-r chose. Of course, you could also just allow people to choose a map themselves and disable the option to choose any map that falls beyond the bounds of the limit proposed earlier. And also, once a player finishes their map, if someone else is still playing, the person who is finished gets put into spectator. Honestly, it's up to the devs on if, when, or how this gets implemented, but I truly hope this idea is taken under heavy consideration, as it would make playing with my friends, and people in general, much easier, and I think most can agree with me on that.

Sorry for the, admittedly, very long list, but I just didn't feel it was necessary to make a bunch of back-to-back posts. Thanks for reading. o/

**Comment by Joehuu** [comment, marked as answer]
There are already discussions/issues opened for these. I would upvote these instead:
1. https://github.com/ppy/osu/discussions/22798
2. https://github.com/ppy/osu/discussions/17759
3.
	1. https://github.com/ppy/osu/discussions/15044
	2. > Next, make it so when you add an element to the game, it immediately appears where the default element would be. That would also make it easier for skinners to line things up.
	3. > Finally, I think it would be a great addition to add an option to hide the editor UI, but still make it so elements can be moved when the UI is hidden.

	I don't think 2 & 3 have a discussion yet, so please make a discussion for each.
4. https://github.com/ppy/osu/issues/5578

In the future, please search for existing feature requests first and upvote those individually instead. It's hard to keep track of a discussion packed with requests like this. I would also split the feature and reasoning into bullet points as it was hard to find the features said here.

**Reply by WalmartWhiteCat** [reply]
Sorry, I was unaware of these already existing discussions. I used the tool to find similar ones, but nothing actually similar came up so I just assumed they hadn't been brought up yet. I'll try to put in more effort to find these discussions when I make suggestions in the future. If you could, would you let me know of a more efficient way to find similar posts than the button which does so for you? It would be helpful.

As for the bullet points. I felt I had laid the ideas out in a way that was easy to understand, but it is a sizeable post so I can understand how it might be messier than I thought it was. I was actually working on more suggestions earlier and did bullet points instead. So yeah, I'll prioritise neatness in later posts.

Thanks for the feedback. I would make a post for 2 and 3 separately as you said, but it seems like they already have discussions, so I won't. Unless I'm reading this wrong and these posts somehow aren't real? If I am wrong, please let me know. If I'm not, have a good day o/

**Reply by Joehuu** [reply]
> If you could, would you let me know of a more efficient way to find similar posts than the button which does so for you?

In the search bar, I search with 1-2 keywords. Like for the first one, I would search for "favorite sort/filter". I also just remember most of these requests as they were duplicated a lot.

> I would make a post for 2 and 3 separately as you said, but it seems like they already have discussions, so I won't. Unless I'm reading this wrong and these posts somehow aren't real? If I am wrong, please let me know. If I'm not, have a good day o/

They aren't. I just quoted your sentences for ~2 & 3~ 3ii and 3iii.

Edit: I meant your second and third feature request of skin editor.

**Reply by WalmartWhiteCat** [reply]
Ah, that makes more sense. Thanks for clearing that up. Alright, I'll make new discussions and make sure I look more thoroughly so I don't add a duplicate discussion. I appreciate the help o7
